shell/testing/tester_main.cc

```diff
@@ -31,6 +32,48 @@
 
 namespace flutter {
 
+class TesterPlatformView : public PlatformView,
+                           public GPUSurfaceSoftwareDelegate {
+ public:
+  TesterPlatformView(Delegate& delegate, TaskRunners task_runners)
+      : PlatformView(delegate, std::move(task_runners)) {}
+
+  std::unique_ptr<Surface> CreateRenderingSurface() override {
+    auto surface = std::make_unique<GPUSurfaceSoftware>(
+        this, true /* render to surface */);
+    FML_DCHECK(surface->IsValid());
+    return surface;
+  }
+
+  sk_sp<SkSurface> AcquireBackingStore(const SkISize& size) override {
+    if (sk_surface_ != nullptr &&
+        SkISize::Make(sk_surface_->width(), sk_surface_->height()) == size) {
+      // The old and new surface sizes are the same. Nothing to do here.
+      return sk_surface_;
+    }
+
+    SkImageInfo info =
+        SkImageInfo::MakeN32(size.fWidth, size.fHeight, kPremul_SkAlphaType,
```

Comment:
This allocation is unnecessary as we are not going to present anything in this surface. We only need it for its canvas. Perhaps we need to rework this interface to be something like `sk_sp<SkCanvas> AcquireBackingStoreCanvas(...)`. In the meantime, you could just create a 1x1 surface to minimize unnecessary allocations during test runs.

Reply:
In fact, you can create a 1x1 surface in the ctor and return the same pointer every `AcquireBackingStore`. Doesn't really matter.

Reply:
Done.

Reply:
I can't do this because of the check in gpu_surface_software here: https://github.com/flutter/engine/blob/master/shell/gpu/gpu_surface_software.cc#L50